sql/scan.go

```diff
@@ -79,7 +79,9 @@ type scanNode struct {
 	kvIndex          int               // current index into the key/value pairs
 	rowIndex         int               // the index of the current row
 	colID            ColumnID          // column ID of the current key
+	valTypes         []parser.Datum    // the index key value types for the current row
 	vals             []parser.Datum    // the index key values for the current row
+	implicitValTypes []parser.Datum    // the implicit value types for unique indexes
```

Comment:
I'm going to add a comment to `decodeKeyVals` as that method is the driver for these members.